### PR TITLE
Add parameterless constructor to LevelDB journals

### PR DESCRIPTION
**Pull request by ortibazar**
Refs #22002

### CODE REVIEW

**Comment by patriknw** [comment]
OK TO TEST

File: akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala [hunk 1]
```diff
@@ -20,9 +20,15 @@ import com.typesafe.config.Config
  *
  * Journal backed by a local LevelDB store. For production use.
  */
-private[persistence] class LeveldbJournal(val config: Config) extends AsyncWriteJournal with LeveldbStore {
+private[persistence] class LeveldbJournal(cfg: Config) extends AsyncWriteJournal with LeveldbStore {
   import LeveldbJournal._
 
+  def this() = this(null)
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: I don't like the null, someone could accidentally use it.
I propose to instead do:

```scala
def this() = this(context.system.settings.config.getConfig("akka.persistence.journal.leveldb"))
```

Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: that would be better, if it works (is context know at that point?)

**Comment by patriknw** [comment]
I verified that multi-jvm tests pass in akka-cluster-sharding

File: akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala (hunk 1 above)
Comment (ortigali) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: In previous PR, I also want to do it like above. But there is no context.

**Comment by patriknw** [comment]
wow that's a compilation error in akka-stream. Can't be this PR so must be something in master

```
[warn] non-nullary method overrides nullary method
[error] /localhome/jenkinsakka/workspace/pr-validator-per-commit-jenkins/akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala:379: forward reference extends over definition of value fromNetTimeout
[error]       fromNetTimeout.reversed // now the bottom flow transforms the exception, the top one doesn't (since that one is "fromNet") 
[error]       ^
[warn] two warnings found
[error] one error found
```

**Comment by ktoso** [comment]
I fixed this on master already, sorry for slipping that in.

**Comment by ktoso** [comment]
PLS BUILD

**Comment by patriknw** [comment]
@ortigali one more thing, can you remove the filter in MiMa.scala for `SharedLeveldbStore.this`. It should now be compatible.

File: akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala (hunk 1 above)
Comment (johanandren) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: What about an empty placeholder config to signal use default?

Comment (johanandren) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]:
```
object LeveldbJournal {
  private val emptyConf = ConfigFactory.empty
}

and then compare instance equality against that inside of actor to know if default should be loaded

Comment (ortigali) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: @johanandren I do not understand your point. Could you give me a hint or sample?

Comment (ortigali) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala]: Sorry, understand now.

**Review by johanandren — APPROVED** [review]
LGTM, thanks for fixing so quickly

**Comment by patriknw** [comment]
Thanks for fixing @ortigali

**Pull request merged**